src/e2e-app/rollup-with-linker.config.js

```diff
@@ -0,0 +1,76 @@
+const node = require('rollup-plugin-node-resolve');
+const {ConsoleLogger, LogLevel} = require("@angular/compiler-cli/src/ngtsc/logging");
+const {NodeJSFileSystem} = require("@angular/compiler-cli/src/ngtsc/file_system");
+const {createEs2015LinkerPlugin} = require('@angular/compiler-cli/linker/babel');
+const {babel} = require('@rollup/plugin-babel');
+const commonjs = require('@rollup/plugin-commonjs');
+const {sync: globSync} = require('glob');
+const {join} = require('path');
+
+/** File system used by the Angular linker plugin. */
+const fileSystem = new NodeJSFileSystem();
+/** Logger used by the Angular linker plugin. */
+const logger = new ConsoleLogger(LogLevel.info);
+/** Babel plugin that runs the Angular linker. */
+const linkerPlugin = createEs2015LinkerPlugin({fileSystem, logger});
+
+/**
+ * Custom rollup plugin that enables bundling of the component examples which are
+ * usually loaded lazily. We cannot use lazy loading here as we want to process
+ * all sources with the linker, and rollup isn't able to process dynamic imports
+ * which are not statically analyzable. We work around this by transforming the
+ * dynamic non-static imports to statically analyzable dynamic imports that can be
+ * processed by rollup. e.g. "import(`@angular/components-examples/${module}/index.js`")
+ * will be transformed into an object that merges exports from all possible `${module}` values.
```

Comment:
It might be worth noting for the future, that if you have `@angular/compiler` loaded at runtime, then partial declarations can be JIT compiled to full definitions, without the need for the linker...

Reply:
I see. good to know! I'm not sure what's best here; but I imagine the JIT compilation slowing down tests a little bit more (if not significantly)